pkg/service/HelmAppService.go

```diff
@@ -1354,3 +1370,143 @@ func (impl HelmAppServiceImpl) UpgradeReleaseWithCustomChart(ctx context.Context
 	}
 	return true, nil
 }
+
+func (impl HelmAppServiceImpl) OCIRegistryLogin(registryCredential *client.OCIRegistryRequest) error {
+	username := registryCredential.Username
+	pwd := registryCredential.Password
+	if registryCredential.RegistryType == REGISTRY_TYPE_ECR {
+		accessKey, secretKey := registryCredential.AccessKey, registryCredential.SecretKey
+		var creds *credentials.Credentials
+
+		if len(registryCredential.AccessKey) == 0 || len(registryCredential.SecretKey) == 0 {
+			sess, err := session.NewSession(&aws.Config{
```

Comment:
are creating new client on every request

Reply:
only in case of ECR registries with IAM role configuration